x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +871,199 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
+
+	// check if the gauge is created
+	clPoolInternalGaugeId, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, clPoolId1.GetId(), epochInfo.Duration)
+	s.Require().NoError(err)
+
+	clPoolId2 := s.PrepareConcentratedPool() // creates internal no-lock gauge id = 2
+
+	// check if the gauge is created
+	_, err = s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, clPoolId2.GetId(), epochInfo.Duration)
+	s.Require().NoError(err)
+
+	// 3. Create external no-lock gauges for CL pools
+	clPoolExternalGaugeIdPool1 := s.CreateNoLockExternalGauges(clPoolId1.GetId(), externalGaugeCoins, s.TestAccs[1], uint64(1))
+	clPoolExternalGaugeIdPool2 := s.CreateNoLockExternalGauges(clPoolId2.GetId(), externalGaugeCoins, s.TestAccs[2], uint64(numEpochsPaidOverGaugeTwo))
+
+	// 4. Create Distribution records to incentivize internal CL no-lock gauges
+	s.IncentivizeInternalGauge([]uint64{clPoolId1.GetId(), clPoolId2.GetId()}, epochInfo.Duration, false)
+
+	// 5. let epoch 1 pass
+	// ******************** EPOCH 1 *********************
+	// let 1 epoch pass
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(epochInfo.Duration))
+	s.App.EpochsKeeper.AfterEpochEnd(s.Ctx, epochInfo.GetIdentifier(), 1)
+
+	// clPoolExternalGaugeIdPool1 expects full because the numEpochPaidOver is 1 for that gagueId
+	//  clPoolExternalGaugeIdPool2 expects half because the numEpochPaidOver is 2 for that gagueId
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool1, 1, externalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool2, 1, halfOfExternalGaugeCoins)
+
+	internalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId1.GetId())
+	s.Require().NoError(err)
+
+	externalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId2.GetId())
+	s.Require().NoError(err)
+
+	// check internalIncentiveCLPool
+	s.Require().Equal(2, len(internalIncentiveClPool))
+	emissionRateForExternalTokens := sdk.NewDecFromInt(sdk.NewInt(defaultExternalGaugeValue)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds()).QuoInt(sdk.NewInt(1000)))
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), externalGaugeDecCoins[0], emissionRateForExternalTokens, internalIncentiveClPool[0])
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), externalGaugeDecCoins[1], emissionRateForExternalTokens, internalIncentiveClPool[1])
+
+	// check externalIncentiveCLPool
+	s.Require().Equal(2, len(externalIncentiveClPool))
+	emissionRateForExternalTokens = sdk.NewDecFromInt(sdk.NewInt(defaultExternalGaugeValue)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds() * 2).QuoInt(sdk.NewInt(1000)))
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), halfOfExternalGaugeDecCoins[0], emissionRateForExternalTokens, externalIncentiveClPool[0])
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), halfOfExternalGaugeDecCoins[1], emissionRateForExternalTokens, externalIncentiveClPool[1])
```

Comment:
So, from this, I can infer that we only expect external gauges to emit. Could you add comments explaining why in-detail as well as why no internal gauges are expected to emit at this point?

Reply:
I believe the reason is that at epoch 1 we first need to mint the tokens and distribute everything to the distr records. As a result, internal gauges only get updated by epoch 2 and not one

Reply:
> I believe the reason is that at epoch 1 we first need to mint the tokens and distribute everything to the distr records. As a result, internal gauges only get updated by epoch 2 and not one

yup this is correct. 

External incentives starts getting distributed at epoch 1, therefore 2 records for each pool
Internal incentives gets distributed at epoch 2, therefore 2records for each pool + 1 internal incentive record